### ISSUE TITLE
buffer size doubling for iocpreactor

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2087]
|[<img alt="ghazel's avatar" src="https://avatars.githubusercontent.com/u/31402?s=50" width="50" height="50">](https://github.com/ghazel)| @ghazel reported|
|-|-|
|Trac ID|trac#2087|
|Type|enhancement|
|Created|2006-09-13 21:19:47Z|

IOCP currently eats 65k per connection for a read buffer. This large memory overhead can be reduced without much cpu overhead by starting small and doubling every time the readbuffer is 100% utilized.

Attached is a patch which accomplishes this

Attachments:

* [buffer.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/64c/64ccc7533460801b124e476834ea5802fd29109b/ff2f183044d8e583e21b25cc1c4258bdd2e14251.diff) (2784 bytes) - added by ghazel on 2006-09-17 00:21:49Z - buffer doubling patch for abstract.py (tab removal)

<details><summary>Searchable metadata</summary>

```
trac-id__2087 2087
type__enhancement enhancement
reporter__ghazel ghazel
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__duplicate duplicate
component__core core
keywords__iocp iocp
time__1158182387000000 1158182387000000
changetime__1205873695000000 1205873695000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

There are no tests in this patch.  It should have some added before it is applied.  As the code is currently factored, it looks like it will be pretty tough to write any unit tests for the behavior change, so the code should probably be refactored too.

**Comment by twisted-trac** [comment]
|[<img alt="ghazel's avatar" src="https://avatars.githubusercontent.com/u/31402?s=50" width="50" height="50">](https://github.com/ghazel)<a name="note_2"></a>|@ghazel commented|
|-|-|

You can run all the standard reactor tests to make sure iocp still works.

Writing a unit test for this seems pointless otherwise. There's no unit test for the old behaviour either.

**Comment by twisted-trac** [comment]
|<img alt="PenguinOfDoom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|PenguinOfDoom set status to `closed`|
|-|-|

Implemented in [#1760](https://github.com/twisted/twisted/issues/1760)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
